wip_pacta_data_qa.Rmd

```diff
@@ -0,0 +1,545 @@
+  ungroup() %>% 
+  mutate(pct_holdings = round(total_holdings / sum(total_holdings),3),
+         market_value_weight = round(sum_market_value / sum(sum_market_value), 3))
+datatable(investors_portfolio_currency_overview)
+
+```
+
+
+## Currencies
+
+### currencies in portfolio and missing currency info
+1. Overview of all currencies present in the portfolio and their respective overall market values (in units of that currency)
+2. Overview of holdings with missing exchange rate information. For successful analysis, exchange rate information is required, so it should be checked if exchange rates can be manually added in these case.
+
+**Notes:**
+
+* Multiple entries of the same ISIN in the missing-table sometimes due to different holding IDs.
+* **market_value_miss_currency** is the aggregated value of the holdings with missing currency information per Investor and PF.
+* **sum_market_value** is the aggregated value of all holdings per Investor and PF.
+* **share_missing_market_value** is the missing share of market value per investor, PF and ISIN due to missing currency info.
+
+**OPEN:** Given the aggregation of the market_value across currencies does not make a lot of sense, a better metric should be found. Adding the currency information to the left_join does not help, as this will lead to only calculating the missing share on a fraction of the original PF's size.
+
+```{r}
+currency_overview <- result_data_qa %>% 
+  filter(investor_name != "Meta Investor") %>% 
+  group_by(currency, has_currency) %>% 
+  summarise(total_per_currency = n(),
+            sum_market_value = round(sum(market_value, na.rm = T))) %>% 
+  ungroup()
+datatable(currency_overview)
+
+
+missing_currency <- result_data_qa %>% 
+  filter((flag == "Missing currency information") &
+           investor_name != "Meta Investor") %>% 
+  group_by(investor_name, portfolio_name, currency, isin, exchange_rate_usd, company_name) %>% 
+  summarise(market_value_miss_currency = round(sum(market_value, na.rm = T))) %>% 
+  ungroup() %>% 
+  left_join(
+    investors_portfolio_overview %>% select(investor_name, portfolio_name, sum_market_value), by = c("investor_name", "portfolio_name")
+    ) %>% 
+  mutate(share_missing_market_value = round(
+    market_value_miss_currency/sum_market_value, 4))
+datatable(missing_currency, rownames = F)
+
+```
+
+## Missing Bloomberg data
+
+### Overview of holdings with missing Bloomberg data
+Check if data for these ISINs can be downloaded from Bloomberg and added to DataStore by Taylor
+Use the .csv file generated by the PACTA_analysis script to pass on the information. This DataTable is supposed to help understand more about individual cases.
+For ISINs that are not in DataStore, check if these are unlisted items (??)
+
+* **share_missing_value** shows the overall impact per ISIN and asset_type
+
+**OPEN:** should financial_sector == 'Other' be filtered out for convenience?
```

Comment:
It could be.. I don't think it matter so much. 
I think directly sorting by "share_missing_value" could be more valuable. 
Also if you can change the .x and .y to reflect what they are that should help. I think the sum_value_usd.x is a duplicate of the value_usd

Reply:
sum_value_usd.x is only a duplicate, if the isin is only held by one investor in one portfolio. otherwise it sums across all these.. grouping by isin. But I will definitely improve the column names.

Reply:
I will also remove the sum_value_usd_total, as I don't see the value in repeating the same value in each row. just needed for the calculation.